pkg/patterns/declarative/reconciler.go

```diff
@@ -143,6 +192,14 @@ func (r *Reconciler) reconcileExists(ctx context.Context, name types.NamespacedN
 	}
 	manifestStr = m
 
+	if r.CollectMetrics() {
+		for _, obj := range objects.Items {
+			if err := globalObjectTracker.addIfNotPresent(obj); err != nil {
```

Comment:
You could move this below the `ns` population logic, and then you can pass in the default namespace into `addIfNotPresent`

Reply:
Thanks. I overlooked that `ns` population logic. I'll do it.